typescript/sdk/src/hook/HyperlaneHookDeployer.ts

```diff
@@ -0,0 +1,150 @@
+    const hookAddress = deployedContractMap.test1.optimismMessageHook.address;
+
+    this.logger(`Setting hook address ${hookAddress} for OptimismISM`);
+    await deployedContractMap.test2.optimismISM.setOptimismHook(hookAddress);
+
+    return deployedContractMap;
+  }
+
+  async deployContracts(
+    chain: ChainName,
+    hookConfig: HookConfig,
+  ): Promise<HyperlaneContracts<HookFactories>> {
+    let optimismISM, optimismMessageHook, testRecipient;
+    if (hookConfig.nativeType === NativeType.ISM) {
+      optimismISM = await this.deployOptimismISM(
+        chain,
+        hookConfig.nativeBridge,
+      );
+      testRecipient = await this.deployTestRecipient(
+        chain,
+        optimismISM.address,
+      );
+      this.logger(
+        `Deployed test recipient on ${chain} at ${this.padToBytes32(
+          testRecipient.address,
+        )}`,
+      );
+
+      return {
+        optimismISM,
+        testRecipient,
+      };
+    } else if (hookConfig.nativeType === NativeType.HOOK) {
+      optimismMessageHook = await this.deployOptimismMessageHook(
+        chain,
+        hookConfig.destinationDomain,
+        hookConfig.nativeBridge,
+        hookConfig.remoteIsm,
+      );
+      return {
+        optimismMessageHook,
+      };
+    }
+    return {};
+  }
+
+  async deployOptimismISM(
+    chain: ChainName,
+    nativeBridge: types.Address,
+  ): Promise<OptimismISM> {
+    const signer = this.multiProvider.getSigner(chain);
+
+    const optimsimISM = await new OptimismISM__factory(signer).deploy(
+      nativeBridge,
+    );
+
+    await this.multiProvider.handleTx(chain, optimsimISM.deployTransaction);
+
+    this.logger(`Deployed OptimismISM on ${chain} at ${optimsimISM.address}`);
+    return optimsimISM;
+  }
+
+  async deployTestRecipient(
+    chain: ChainName,
+    ism: types.Address,
+  ): Promise<TestRecipient> {
+    const signer = this.multiProvider.getSigner(chain);
+
+    const testRecipient = await new TestRecipient__factory(signer).deploy();
+
+    await this.multiProvider.handleTx(chain, testRecipient.deployTransaction);
+
+    await testRecipient.setInterchainSecurityModule(ism);
+    return testRecipient;
+  }
+
+  async deployOptimismMessageHook(
+    chain: ChainName,
+    destinationDomain: number,
+    nativeBridge: types.Address,
+    optimismISM: types.Address,
+  ): Promise<OptimismMessageHook> {
+    const signer = this.multiProvider.getSigner(chain);
+
+    const optimismMessageHook = await new OptimismMessageHook__factory(
+      signer,
+    ).deploy(destinationDomain, nativeBridge, optimismISM);
+
+    await this.multiProvider.handleTx(
+      chain,
+      optimismMessageHook.deployTransaction,
+    );
+    this.logger(
+      `Deployed OptimismMessageHook on ${chain} at ${optimismMessageHook.address}`,
+    );
+    return optimismMessageHook;
+  }
+
+  // Function to pad a hexadecimal string to 32 bytes
+  padToBytes32(hexString: string): string {
```

Comment:
use `utils.addressToBytes32` and vice versa
see my other comments about address though